python/cudf/cudf/core/column/numerical.py

```diff
@@ -404,10 +405,30 @@ def _process_values_for_isin(
         self, values: Sequence
     ) -> Tuple[ColumnBase, ColumnBase]:
         lhs = cast("cudf.core.column.ColumnBase", self)
-        rhs = as_column(values, nan_as_null=False)
-
-        if isinstance(rhs, NumericalColumn):
-            rhs = rhs.astype(dtype=self.dtype)
+        try:
+            rhs = as_column(values, nan_as_null=False)
+        except (MixedTypeError, TypeError) as e:
```

Comment:
Just curious: Does one of the tested cases produce/catch a `MixedTypeError`? I see where the (`MixedTypeError` will be produced)

Reply:
Good catch, Updated test suite to cover this case.